js/state/siteUtil.js

```diff
@@ -25,22 +25,20 @@ const isBookmarkFolder = (tags) => {
     tags && typeof tags !== 'string' && tags.includes(siteTags.BOOKMARK_FOLDER)
 }
 
-/**
- * Obtains the index of the location in sites
- *
- * @param sites The application state's Immutable sites list
- * @param siteDetail The siteDetails entry to get the index of
- * @param tags Tag for siteDetail (ex: bookmark). Folders are searched differently than other entries
- * @return index of the siteDetail or -1 if not found.
- */
-module.exports.getSiteIndex = function (sites, siteDetail, tags) {
-  if (!sites || !siteDetail) {
-    return -1
+module.exports.getSiteKey = function (siteDetail, tags) {
+  if (!siteDetail) {
+    return null
   }
-  if (isBookmarkFolder(tags)) {
-    return sites.findIndex((site) => isBookmarkFolder(site.get('tags')) && site.get('folderId') === siteDetail.get('folderId'))
+  const folderId = siteDetail.get('folderId')
```

Comment:
with this refactor, the tags field really isn't needed (which is nice).  Because you can determine if it's a folder or not by the `folderId`.  It might clean things up to remove it